nomad/structs/structs_test.go

```diff
@@ -6024,3 +5917,30 @@ func TestTaskGroup_validateScriptChecksInGroupServices(t *testing.T) {
 		require.Nil(t, mErrOK)
 	})
 }
+
+func requireMultierrorNotContaining(t *testing.T, err error, unexpected ...string) {
+	t.Helper()
+	require.Error(t, err)
+	mErr, ok := err.(*multierror.Error)
+	require.True(t, ok)
+
+	var found []string
+	for _, unex := range unexpected {
+		if strings.Contains(mErr.Error(), unex) {
+			found = append(found, unex)
+		}
+	}
+
+	require.Empty(t, found, "found unexpected error(s): %v", found)
+}
+
+func requireMultierrorContaining(t *testing.T, err error, expected ...string) {
```

Comment:
This function doesn't seem to actually rely on `err` being a multierror either.

Also, I noticed the original tests assert the order of errors as well and error content of each individual error message, while this helper relaxes the requirement and does a simple contain check across all errors.  Is that intentional relaxation?  I don't know these tests well enough to evaluate impact well.

Reply:
The ordering is meant to be lost, but I will bring back the assertions on the individual errors.